src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -34,11 +34,25 @@ public static class Helpers
                 "void", "partial", "yield", "where"
             };
 
+        private static int numSafeIden = 0;
         public static string SafeIdentifier(string id)
         {
             if (id.All(char.IsLetterOrDigit))
                 return Keywords.Contains(id) ? "@" + id : id;
-            return new string(id.Select(c => char.IsLetterOrDigit(c) ? c : '_').ToArray());
+            
+            id = new string(id.Select(c => char.IsLetterOrDigit(c) ? c : '_').ToArray());
+
+            if (id.Replace("_", "") == string.Empty)
```

Comment:
string.Replace creates a new string which is expensive. The fastest option here is:
if (id.All(c => c == '_'))